File: code_samples/scala/async_coroutines.scala
/**
 * Async: Coroutines
 * AI/ML Training Sample
 */

class Coroutines {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object CoroutinesExample {
  def main(args: Array[String]): Unit = {
    val instance = new Coroutines()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/csharp/async_coroutines.cs
using System;

/// <summary>
/// Async: Coroutines
/// AI/ML Training Sample
/// </summary>
public class Coroutines
{
    private string data;
    
    public Coroutines()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Coroutines();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}
